### PR TITLE
[FX] Ensure BC coverage for all of torch.fx.passes

### PR DESCRIPTION
**Pull request by jamesr66a**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #65081
* #65183

`torch.fx.passes` is not imported by `torch.fx` (for circular dependency reasons) so it was not getting picked up by the BC API surface test (more specifically: only the modules in passes that were used by things outside of passes were getting picked up). This makes `passes` into a bona fide module by adding an `__init__.py` and adds BC decorators to all of the contained public members (conservatively setting them to False).

Differential Revision: [D30967428](https://our.internmc.facebook.com/intern/diff/D30967428)

### CODE REVIEW

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

File: torch/fx/passes/graph_drawer.py [hunk 1]
```diff
@@ -43,6 +44,7 @@
 }
 
 
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/graph_drawer.py:47]: cc @khabinov FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

File: torch/fx/passes/net_min_base.py [hunk 2]
```diff
@@ -21,22 +22,23 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/net_min_base.py:25]: cc @842974287 FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

File: torch/fx/passes/operator_support.py [hunk 3]
```diff
@@ -2,10 +2,11 @@
 
 import torch
 import torch.fx
+from torch.fx._compatibility import compatibility
 
 from .tools_common import get_node_target, CALLABLE_NODE_OPS
 
-
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/operator_support.py:9]: cc @842974287 FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

File: torch/fx/passes/split_utils.py [hunk 4]
```diff
@@ -5,8 +5,10 @@
 import torch.nn as nn
 from torch.fx.graph import map_arg
 from .tools_common import NodeList, NodeSet
+from torch.fx._compatibility import compatibility
 
 
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/split_utils.py:11]: cc @842974287 FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

File: torch/fx/passes/splitter_base.py [hunk 5]
```diff
@@ -63,6 +64,7 @@ def __init__(self):
 
 
 # TODO: this can probably be optimized
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/splitter_base.py:67]: cc @842974287 FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

File: torch/fx/passes/tools_common.py [hunk 6]
```diff
@@ -13,7 +14,7 @@
 Names = List[str]
 CALLABLE_NODE_OPS = {"call_module", "call_function", "call_method"}
 
-
+@compatibility(is_backward_compatible=False)
```
Comment (jamesr66a) [inline torch/fx/passes/tools_common.py:17]: cc @842974287 FYI I'm conservatively setting backward-compatibility for the APIs in the file to `False`. Please feel free to flip this to `True` if/when you feel comfortable providing stability guarantees for the API.

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

**Review by khabinov — APPROVED** [review]

**Review by khabinov — APPROVED** [review]

**Review by khabinov — APPROVED** [review]

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

**Comment by jamesr66a** [comment]
@jamesr66a has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30967428).

**Pull request closed without merging**